app/code/core/Mage/SalesRule/Model/Resource/Rule/Collection.php

```diff
@@ -79,7 +79,7 @@ public function setValidationFilter($websiteId, $customerGroupId, $couponCode =
             $select = $this->getSelect();
 
             $connection = $this->getConnection();
-            if (strlen($couponCode)) {
+            if ($couponCode !== '') {
```

Comment:
if couponCode is null then this check is true, isn't it?
I would rather  use `!empty($couponCode) `

Reply:
Or alternatively we could add `is_string($couponCode)` to the conditional.

Reply:
@tmotyl right. Not sure what ive tested yesterday?!?